src/robust_provider/provider.rs

```diff
@@ -532,4 +546,127 @@ mod tests {
+        let block = stream.next().await.unwrap()?;
+        assert_eq!(1, block.number());
+
+        for _ in 0..30 {
+            ws_provider_2.anvil_mine(Some(10), None).await?;
+            let _ = stream.next().await.unwrap()?;
+            sleep(Duration::from_millis(100)).await;
+            // Mine on primary - should reconnect and receive from primary
+            ws_provider_1.anvil_mine(Some(1), None).await?;
+        }
+
+        let block = stream.next().await.unwrap()?;
+        assert_eq!(31 + 1, block.number());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_subscription_cycles_through_multiple_fallbacks() -> anyhow::Result<()> {
+        let anvil_1 = Anvil::new().try_spawn()?;
+        let ws_provider_1 =
+            ProviderBuilder::new().connect(anvil_1.ws_endpoint_url().as_str()).await?;
+
+        let anvil_2 = Anvil::new().try_spawn()?;
+        let ws_provider_2 =
+            ProviderBuilder::new().connect(anvil_2.ws_endpoint_url().as_str()).await?;
+
+        let anvil_3 = Anvil::new().try_spawn()?;
+        let ws_provider_3 =
+            ProviderBuilder::new().connect(anvil_3.ws_endpoint_url().as_str()).await?;
+
+        let robust = RobustProviderBuilder::fragile(ws_provider_1.clone())
+            .fallback(ws_provider_2.clone())
+            .fallback(ws_provider_3.clone())
+            .subscription_timeout(Duration::from_millis(500))
+            .build()
+            .await?;
+
+        let subscription = robust.subscribe_blocks().await?;
+
+        let mut stream = subscription.into_stream();
+
+        ws_provider_1.anvil_mine(Some(1), None).await?;
+        let block = stream.next().await.unwrap()?;
+        assert_eq!(1, block.number());
+
+        // simulate ws stream gone via no blocks mined > sub timeout
+        sleep(Duration::from_millis(600)).await;
+
+        ws_provider_2.anvil_mine(Some(1), None).await?;
+        let block = stream.next().await.unwrap()?;
+        assert_eq!(1, block.number());
+
+        // simulate ws stream gone via no blocks mined > sub timeout
+        sleep(Duration::from_millis(600)).await;
+
+        ws_provider_3.anvil_mine(Some(1), None).await?;
+        let block = stream.next().await.unwrap()?;
+        assert_eq!(1, block.number());
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_subscription_fails_with_no_fallbacks() -> anyhow::Result<()> {
+        let anvil = Anvil::new().try_spawn()?;
+        let ws_provider = ProviderBuilder::new().connect(anvil.ws_endpoint_url().as_str()).await?;
+
+        let robust = RobustProviderBuilder::fragile(ws_provider.clone())
+            .subscription_timeout(Duration::from_millis(500))
+            .build()
+            .await?;
+
+        let mut subscription = robust.subscribe_blocks().await?;
+
+        // simulate ws stream gone via no blocks mined > sub timeout
+        ws_provider.anvil_mine(Some(1), None).await?;
+        let _block = subscription.recv().await?;
```

Comment:
```suggestion
        ws_provider.anvil_mine(Some(1), None).await?;
        let _ = subscription.recv().await?;
```
nit: forgot to delete comment?